### ISSUE TITLE
Cpanel redirect /cgi-sys/defaultwebpage.cgi mean website is DOWN, but uptime-kuma is UP

### ISSUE THREAD

**Issue by vulieumang** [louislam/uptime-kuma#3189]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

When website redirect to /cgi-sys/defaultwebpage.cgi mean website is DOWN, but uptime-kuma is UP
<img width="1115" alt="image" src="https://github.com/louislam/uptime-kuma/assets/13487753/c47dbbf4-2337-473c-a3a1-b0ad8b7129c1">
is had been from cpanel hosting when stop or something happen.

I had edit max-redirect to 0 but still up
<img width="669" alt="image" src="https://github.com/louislam/uptime-kuma/assets/13487753/2b1ff96a-c70b-4498-acc3-3d2bb3c39bf8">

### ✔️ Solution

I want have any way to check if website redirect to cgi-sys/defaultwebpage.cgi  and alert DOWN, 
We can contact to hosting provider and fix it

**Comment by CommanderStorm** [comment]
Cannot reproduce this. 
The site https://nav.tum.sexy (redirecting to https://nav.tum.de with a permanent redirect) is displayed like this.
![image](https://github.com/louislam/uptime-kuma/assets/26258709/a867267f-93e8-4768-b78f-c21fbdecd7c0)

Could you share further details on what kind of status code is delivered if the website is down?

**Comment by CommanderStorm** [comment]
if the down status code falls in the 200 range, I would recomend using the `HTTP - Keyword` monitor instead:
![image](https://github.com/louislam/uptime-kuma/assets/26258709/7ad57722-8803-42eb-9822-918e40a23613)

**Comment by vulieumang** [comment]
@CommanderStorm thank you so much for reply,
my website is https://giavihuyentrang.com/
is redirect to https://giavihuyentrang.com/cgi-sys/defaultwebpage.cgi

**Comment by CommanderStorm** [comment]
The site answers with a 
![image](https://github.com/louislam/uptime-kuma/assets/26258709/1f4db1c6-361d-4982-9ca2-4bdc98782a71)

=> given that this is a 200 exit code, there is nothing we can do.
HTML parsing and finding this `meta`-tag is something that should be out of scope for an uptime monitor...
Do you agree?

I would suggest to monitor for this via the Keyword uptime monitor as described above.

**Comment by vulieumang** [comment]
@CommanderStorm thank you so much, I very surprise when you care my case.
Can you help me setup keyword for this case?
I want both normal and this case in all my domain, because I using cpanel in almost website

**Comment by CommanderStorm** [comment]
![image](https://github.com/louislam/uptime-kuma/assets/26258709/a753dd41-b753-4360-b456-f9733ccee282)
![image](https://github.com/louislam/uptime-kuma/assets/26258709/042f23fe-7ab2-4942-9f68-d55a091bb664)

**Comment by vulieumang** [comment]
@CommanderStorm thank you so much,
I think website using wordpress and Cpanel is very large.
Can you consider an feature like Cpanel checkbox, when checkbox is true, it will check 
<meta http-equiv="refresh" content="0;URL=/cgi-sys/defaultwebpage.cgi">
and return DOWN

**Comment by CommanderStorm** [comment]
> wordpress and Cpanel is very large

I dont believe that this meta tag is widely used.
[Wordpress does deliver 301/309 redirects](https://themeisle.com/blog/301-redirect-wordpress/).
[Cpanel also does support these redirects](https://docs.cpanel.net/cpanel/domains/redirects/).

I have looked into if adding a regex to support `http-equiv="refresh"` would be possible.
I have come to the conclusion that this would be very messy.
A Regex would need to handle:
- [whitespace anywere](https://stackoverflow.com/questions/1370025/regex-for-http-equiv-refresh-meta-tag)
- case-indifference, except in the url
- supporting `'` and `"`
- supporting both orderings of `http-equiv="refresh"` and `content`

=> Adding such a feature would also be breaking for other users.
=> Adding such a feature would add much more complexity around here https://github.com/louislam/uptime-kuma/blob/f2633a5d0163b898b8214bb1308e76e359a425c6/server/model/monitor.js#L287
=> given that axios (the http library) does not support this legacy feature from the netscape days, I don't think we should add an extra component to address this.

**Comment by vulieumang** [comment]
@CommanderStorm thank you so much for your research.
But please find a way, if have 50 site on cpanel, if do not have any option, I must re add 50 website duplicate,
it mean, every website cpanel, we must add twice per address

**Comment by CommanderStorm** [comment]
> if have 50 site on cpanel, if do not have any option, I must re add 50 website duplicate, it mean, every website cpanel, we must add twice per address

I don't fully understand what you mean. Editing the monitor type does not require recreation, nor duplicates
![image](https://github.com/louislam/uptime-kuma/assets/26258709/9a0e108b-64a9-4ccc-87b9-b32643f41c9a)

**Comment by vulieumang** [comment]
if we using keyword
When uptimekuma read my site it will find keyword "<meta http-equiv="refresh" content="0;URL=/cgi-sys/defaultwebpage.cgi">"
if not have it will return up it is ok.
but if my site down for other reason, keyword "meta refresh"  will not found and uptimekuma will return UP

May I misunderstood?

**Comment by CommanderStorm** [comment]
I don't understand what you are trying to communicate.

You said that `http-equiv="refresh"` being in the requst is how the site indicates it is down.
=> What other states are there?
=> One healthcheck should monitor for exactly one failiour case.
      Otherwise, you won't know what you need to debug.

**Comment by vulieumang** [comment]
sorry, I will send example, it will be easy.

here is my website but I had point to other server down and result still UP
`https://giavihuyentrang.com/`
<img width="1205" alt="image" src="https://github.com/louislam/uptime-kuma/assets/13487753/6cf74bac-cf46-4d7f-9072-fb77510996bb">
<img width="1063" alt="image" src="https://github.com/louislam/uptime-kuma/assets/13487753/a2eac05a-88bf-4453-81e2-88737886c882">

It mean, keyword only work if server is working, it server not working, It will be wrong

**Comment by CommanderStorm** [comment]
Your host not responding should be checked via a different health check.

**Comment by louislam** [comment]
I think you should not check `<meta http-equiv="refresh"` in the first place, let me suggest a simpler way.

You should specific positive words which are existing on your homepage. Since your website is inaccessible right now, I assume that there are words - `Hello! This is my website!`.

Monitor Config:
URL: https://giavihuyentrang.com
Monitor Type: HTTPS(s) - Keyword
Keyword: `Hello! This is my website!`

In this way:
UP = Found the keyword
DOWN = Other possibilities (Include Cloudflare error and cpanel error)

**Comment by vulieumang** [comment]
@CommanderStorm 
Yes, I think we using uptime for health check and if user using Cpanel, user must using 2 type http normal and  http keyword, it mean every website we have duplicate, and it so take time :(, I wish have option for health check and Cpanel in my case.

@louislam thank you hero, I think your way is a good way, But with 50 site it will be more complex :(, and hard to check website we not owner.

 I understand that only way to do now.
thank both :)
Can I contribute a feature, a checkbox below advance option name is "Cpanel support" and it will check body, I don't know how to do, but I will try

**Comment by vulieumang** [comment]
@CommanderStorm hello, If I using keyword, result do not have ping response :( so sad
<img width="1170" alt="image" src="https://github.com/louislam/uptime-kuma/assets/13487753/f1c22ddd-7a75-4dc3-bcd2-c16caba79fc2">

**Comment by CommanderStorm** [comment]
I do not understand what you are trying to telling me.

I have already send a working screenshot.
Louis has also said that monitoring for positive keywords is probably better approach than negative keywords, which I agree.

Btw: You might want to use deepl.com to make your points clearer.
